xapian-core/include/xapian/weight.h

```diff
@@ -1082,6 +1082,80 @@ class XAPIAN_VISIBILITY_DEFAULT PL2Weight : public Weight {
     double get_maxextra() const;
 };
 
+/// Xapian::Weight subclass implementing the PL2+ probabilistic formula.
+class XAPIAN_VISIBILITY_DEFAULT PL2PlusWeight : public PL2Weight {
+
+  public:
+    /** Construct a PL2PlusWeight.
+     *
+     *  @param c  A non-negative and non zero parameter controlling the extent
+     *		  of the normalization of the wdf to the document length. The
+     *		  default value of 1 is suitable for longer queries but it may
+     *		  need to be changed for shorter queries. For more information,
+     *		  please refer to Gianni Amati's PHD thesis titled
+     *		  Probabilistic Models for Information Retrieval based on
+     *		  Divergence from Randomness.
+     *  
+     *  @param delta  A parameter for pseudo tf value to control the scale
+     *                of the tf lower bound. Delta(δ) should be a positive
+     *                real number or in other words, non-negative and non-zero.
+     *                It can be tuned for e.g from 0.1 to 1.5 in increments of
+     *                0.1 or so. Experiments have shown that PL2+ works effectivoly
+     *                across collections with a fixed δ = 0.8 (default 0.8)
+     */
+    PL2PlusWeight(double c, double delta);
```

Comment:
The formula for `PL2` uses `c` for this parameter, so I think that's probably the best name for it here too.  The documentation comment for it can (and does) explain its purpose.  If we called it something like `length_normalisation_extent` we'd then also need to note that this is parameter c in the PL2 formula...